flytekit/core/map_task.py

```diff
@@ -1,72 +1,215 @@
-from typing import Any, Optional
+"""
+Flytekit map tasks specify how to run a single task across a list of inputs. Map tasks themselves are constructed with
+a reference task as well as run-time parameters that limit execution concurrency and failure tolerations.
+"""
+import os
+from typing import Any, Dict, List, Optional, Type
 
-from flytekit.core.base_task import PythonTask, TaskMetadata
+from flytekit.common.tasks.raw_container import _get_container_definition
+from flytekit.core.base_task import PythonTask
+from flytekit.core.context_manager import ExecutionState, FlyteContext, SerializationSettings
 from flytekit.core.interface import transform_interface_to_list_interface
+from flytekit.core.python_function_task import PythonFunctionTask, get_registerable_container_image
+from flytekit.models.array_job import ArrayJob
+from flytekit.models.interface import Variable
+from flytekit.models.task import Container
 
 
 class MapPythonTask(PythonTask):
     """
-    TODO We might need a special entrypoint to start execution of this task type as there is possibly no instance of this
-    type and it needs to be dynamically generated at runtime. We can easily generate it by passing it the actual task
-    that is to be generated.
-
-    To do this we might have to give up on supporting lambda functions initially
+    A MapPythonTask defines a :py:class:`flytekit.PythonTask` which specifies how to run
+    an inner :py:class:`flytekit.PythonFunctionTask` across a range of inputs in parallel.
+    TODO: support lambda functions
     """
 
-    def __init__(self, tk: PythonTask, metadata: Optional[TaskMetadata] = None, **kwargs):
-        collection_interface = transform_interface_to_list_interface(tk.python_interface)
-        name = "mapper_" + tk.name
-        self._run_task = tk
+    def __init__(
+        self,
+        python_function_task: PythonFunctionTask,
+        concurrency: int = None,
+        min_success_ratio: float = None,
+        **kwargs,
+    ):
+        """
+        :param task_function: This argument is implicitly passed and represents the repeatable function
```

Comment:
should this match with arg `python_function_task`?

Reply:
done